src/app/workflow/dag/cwl-viewer/cwl-viewer.component.ts

```diff
@@ -0,0 +1,90 @@
+/*
+ *    Copyright 2018 OICR
+ *
+ *    Licensed under the Apache License, Version 2.0 (the 'License');
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an 'AS IS' BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+import { Component, Input, OnInit } from '@angular/core';
+import { CwlViewerDescriptor, CwlViewerService } from './cwl-viewer.service';
+import { WorkflowVersion } from '../../../shared/swagger/model/workflowVersion';
+
+@Component({
+  selector: 'app-cwl-viewer',
+  templateUrl: './cwl-viewer.html',
+  providers: [CwlViewerService],
+  styleUrls: ['./cwl-viewer.scss']
+})
+
+export class CwlViewerComponent implements OnInit {
+
+  @Input() workflow: any;
+  @Input() set selectedVersion(value: WorkflowVersion) {
+    if (value != null) {
+      this._selectedVersion = value;
+      this.onChange();
+    }
+  }
+  @Input() set refresh(value: any) {
+    this.resetZoom();
+  }
+  @Input() expanded: boolean;
+
+  public cwlViewerError = false;
+  public cwlViewerPercentageZoom = 100;
+  public cwlViewerDescriptor: CwlViewerDescriptor;
+  public loading = false;
+
+  private _selectedVersion;
```

Comment:
https://angular.io/guide/styleguide Style 03-04

Avoid prefixing private properties and methods with an underscore.

Reply:
Changed